imports/plugins/core/checkout/client/templates/checkout/login/login.js

```diff
@@ -17,10 +17,11 @@ Template.checkoutLogin.helpers({
     const cart = Cart.findOne();
     if (cart && cart.workflow) {
       const currentStatus = cart.workflow.status;
-      const guestUser = Reaction.hasPermission("guest", Meteor.user());
-      const anonUser = Roles.userIsInRole("anonymous", Meteor.user(), Reaction.getShopId());
+      const guestUser = Reaction.hasPermission("guest", Meteor.userId(), Reaction.getShopId());
+      const anonUser = Reaction.hasPermission("anonymous", Meteor.userId(), Reaction.getShopId());
 
-      if (currentStatus !== self.template && guestUser === true && anonUser === false) {
+      // when user is guest, they are also anonymous.
```

Comment:
Can you describe this better? i.e to explain what the check below it does?

Seeing `when user is guest, they are also anonymous` doesn't particular help a reader to understand why we do this check.

Reply:
@impactmass I have fixed the changes you requested, and updated the comments. Please confirm it works well now